### PR TITLE
Fix systemd-resolved conflict

### PR DESCRIPTION
**Pull request by axelrtgs**
Fixes https://github.com/sous-chefs/dnsmasq/issues/35

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: recipes/default.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 if platform?('ubuntu') && node['platform_version'] >= '18.04'
+  replace_or_add 'Fix systemd-resolved conflict' do
+    path '/etc/systemd/resolved.conf'
+    pattern 'DNSStubListener=*'
+    line 'DNSStubListener=no'
+  end
+
   service 'systemd-resolved' do
-    action :stop
+    action :restart
```
Comment (qubitrenegade) [inline recipes/default.rb]: As written, this will cause `systemd-resolved` to restart every invocation of `chef-client`

Comment (axelrtgs) [reply recipes/default.rb]: you are 100% correct, i will fix it now and push a new commit.

**Comment by axelrtgs** [comment]
Thanks for the review, I have fixed it. It was not my intention to restart on every run just once to pick up the change. Glad you caught this in the code review. :)

**Comment by damacus** [comment]
As per discussion in Slack, is there a way we can manage the whole file rather than part of this file?

**Comment by axelrtgs** [comment]
@Xorima Small nudge and thanks for having a look.

**Comment by axelrtgs** [comment]
One suggestion was to have the user manage this, i think managing the whole resolved config would conflict with users managing their resolved using the appropriate cookbook and be out of scope for the dnsmasq cookbook

**Comment by KrisShannon** [comment]
I'm not totally sure if `systemd-resolved` has always been able to use drop-in's but it definitely can in v232 when `DNSStubListener` was introduced.

Why not create `/etc/systemd/resolved.conf.d/dnsmasq.conf` with the content:
```
[Resolve]
DNSStubListener=no
```

**Comment by axelrtgs** [comment]
> I'm not totally sure if `systemd-resolved` has always been able to use drop-in's but it definitely can in v232 when `DNSStubListener` was introduced.
> 
> Why not create `/etc/systemd/resolved.conf.d/dnsmasq.conf` with the content:
> 
> ```
> [Resolve]
> DNSStubListener=no
> ```

Thanks for this, i made the changes to use this method which is the best to solve this issue IMO. I appreciate the help. Hopefully now @damacus will approve and we can get this merged and released.

**Comment by damacus** [comment]
Thanks everyone! This we a great conversation.

Hopefully I'll get this shipped this evening

**Pull request merged**